### ISSUE TITLE
[RFE] Introduce a new option to parametrize how krb5 principals are created

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1549]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1545
- Created at 2012-09-26 23:19:35 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed as Invalid
- Assigned to nobody
---

Users sometimes run into the situation where the user principal attribute in LDAP is populated, but points to a different realm than the user wants to utilize, typically a lab vs. production mismatch.

We should introduce a new option that would allow us to override the principal attribute and parametrize it, perhaps with similar expansions that are allowed in the ccname template. 

Usually the user would just use user@SOMEOTHERREALM.COM anyway.

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-09-27 15:26:19](https://pagure.io/SSSD/sssd/issue/1545#240913)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.10 beta
priority: major => minor
rhbz:  => todo
summary: Introduce a new option to parametrize how krb5 principals are created => [RFE] Introduce a new option to parametrize how krb5 principals are created
type: defect => enhancement

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-12-20 23:33:42](https://pagure.io/SSSD/sssd/issue/1545#240914)**

Fields changed

selected:  => Not need

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-01-02 15:32:09](https://pagure.io/SSSD/sssd/issue/1545#240915)**

Moving tickets that are not a priority for SSSD 1.10 into the next release.

milestone: SSSD 1.10 beta => SSSD 1.11 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-26 18:49:56](https://pagure.io/SSSD/sssd/issue/1545#240916)**

Fields changed

mark:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-10 17:06:21](https://pagure.io/SSSD/sssd/issue/1545#240917)**

Fields changed

changelog:  => 
design:  => 
design_review:  => 0
fedora_test_page:  => 
milestone: SSSD 1.13 beta => SSSD 1.13 backlog
priority: minor => trivial
review:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-12 20:36:05](https://pagure.io/SSSD/sssd/issue/1545#240918)**

Mass-moving tickets not planned for any immediate release and re-setting priority.

milestone: SSSD 1.13 backlog => SSSD Deferred
priority: trivial => major

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-17 11:29:35](https://pagure.io/SSSD/sssd/issue/1545#240919)**

This was not needed for four years, I suggest we just close the ticket.

review: 0 => 1
sensitive:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-17 11:30:41](https://pagure.io/SSSD/sssd/issue/1545#240920)**

Fields changed

blockedby:  => 2887

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-17 11:31:02](https://pagure.io/SSSD/sssd/issue/1545#240921)**

Fields changed

blockedby: 2887 =>

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-25 09:45:36](https://pagure.io/SSSD/sssd/issue/1545#240922)**

Fields changed

resolution:  => wontfix
status: new => closed

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 15:03:17](https://pagure.io/SSSD/sssd/issue/1545#240923)**

**Metadata Update from @jhrozek**:
- Issue marked as depending on: #2887
- Issue set to the milestone: SSSD Patches welcome

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-03 11:57:42](https://pagure.io/SSSD/sssd/issue/1545#429235)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review adjusted to on (was: 1)
- Custom field sensitive reset
- Custom field testsupdated reset
